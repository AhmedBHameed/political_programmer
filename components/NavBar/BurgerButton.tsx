import {MenuOutlined} from '@ant-design/icons';

import {Button} from 'antd';
import styled from 'styled-components';

const BurgerIcon = styled(MenuOutlined)`
  margin-top: 0.2rem;
  font-size: 2rem;
`;

const StyledButton = styled(Button)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  border: 0;
`;

interface BurgerButtonProps {
  onClick?: React.MouseEventHandler<HTMLElement>;
}

const BurgerButton: React.FC<BurgerButtonProps> = ({onClick}) => (
  <StyledButton
    data-testid="burger-button"
    icon={<BurgerIcon style={{fontSize: '1.4rem'}} />}
    onClick={onClick}
    type="text"
  />
);

export default BurgerButton;
