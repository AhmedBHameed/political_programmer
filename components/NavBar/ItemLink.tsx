import {Menu} from 'antd';
import React from 'react';
import Link from 'next/link';
import styled from 'styled-components';
import {menuItemStyle, navItemFontSize} from './shared/sharedStyles';
import {mediumGathomFontStyle} from '../shared/sharedStyles';

const StyledLink = styled(Link)`
  ${mediumGathomFontStyle};
`;

const StyledAntdMenuItem = styled(Menu.Item)`
  ${menuItemStyle}
  ${navItemFontSize}
`;
interface ItemLinkProps {
  to: string;
  label: string;
  icon?: React.ReactNode;
  onClick?: () => void;
}

const ItemLink: React.FC<ItemLinkProps> = (props) => {
  const {to, label, ...reset} = props;
  return (
    <StyledAntdMenuItem {...reset} data-testid="navbar-item">
      <StyledLink href={to}>{label}</StyledLink>
    </StyledAntdMenuItem>
  );
};

export default ItemLink;
