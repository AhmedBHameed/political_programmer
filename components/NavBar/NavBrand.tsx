import styled from 'styled-components';
import React from 'react';
import Brand from '../Brand/Brand';
import useNavigateToBlog from './hooks/navigateToBlogHook';

const BrandContainer = styled.div`
  height: 100%;
  overflow: hidden;
  cursor: pointer;
`;

interface NavBrandProps {
  className?: string;
}

const NavBrand: React.FC<NavBrandProps> = ({className}) => {
  const {goToBlog} = useNavigateToBlog();

  return (
    <BrandContainer className={className} onClick={goToBlog}>
      <Brand />
    </BrandContainer>
  );
};

export default NavBrand;
