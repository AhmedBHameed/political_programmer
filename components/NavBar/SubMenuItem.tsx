import {Menu} from 'antd';
import React from 'react';
import {navItemFontSize} from './shared/sharedStyles';
import styled from 'styled-components';

interface SubMenuProps {
  label: string;
  icon?: React.ReactNode;
  onClick?: () => void;
}

const StyledLabel = styled.span`
  font-size: 0.9rem;
`;

const SubMenuItem: React.FC<SubMenuProps> = (props) => {
  const {label, ...reset} = props;

  return (
    <Menu.Item {...reset} data-testid="navbar-item">
      <StyledLabel>{label}</StyledLabel>
    </Menu.Item>
  );
};

export default SubMenuItem;
