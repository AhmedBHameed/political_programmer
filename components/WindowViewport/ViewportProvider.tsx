import {createContext, useEffect, useState} from 'react';

import devices from '../../styles/mediaQueries';

interface ViewportSettings {
  isTabletOrLarger?: boolean;
}

export const ViewportContext = createContext<ViewportSettings>({});

const ViewportProvider: React.FC = ({children}) => {
  const [isTabletOrLarger, setIsTabletOrLarger] = useState<boolean | undefined>(
    undefined
  );

  useEffect(() => {
    const match = window.matchMedia(devices.mdMediaQuery);
    function handleResize() {
      setIsTabletOrLarger(match.matches);
    }

    match.addEventListener('change', handleResize);
    // Call handler right away so state gets updated with initial window size
    handleResize();

    return () => match.removeEventListener('change', handleResize);
  }, []);

  return (
    <ViewportContext.Provider value={{isTabletOrLarger}}>
      {isTabletOrLarger !== undefined && children}
    </ViewportContext.Provider>
  );
};

export default ViewportProvider;
