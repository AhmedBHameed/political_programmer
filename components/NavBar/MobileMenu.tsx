import {SnippetsOutlined, UserOutlined} from '@ant-design/icons';
import {Drawer} from 'antd';
import React from 'react';

import styled from 'styled-components';
import NAVIGATION_ROUTES from '../../config/NavigationRoutes';
import {useTranslation} from '../shared/hooks/useTranslation';
import ItemLink from './ItemLink';
import SubMenuItem from './SubMenuItem';
import NavBrand from './NavBrand';
import {ProfileMenuTitle} from './shared/ProfileMenuTitle';
import {
  spaceXSmall,
  StyledAntdSubMenu,
  StyledAntdMenu,
} from './shared/sharedStyles';

const StyledDrawer = styled(Drawer)`
  .ant-drawer-header {
    ${spaceXSmall}
  }
  .ant-drawer-body {
    padding: 2.4rem 0;
  }
`;

const StyledNavBrand = styled(NavBrand)`
  margin: 0.7rem 0;
`;

interface MenuProps {
  open: boolean;
  activeIndex: string;
  onClose?: () => void;
  onLogout?: () => void;
  onLogin?: () => void;
}

const Menu: React.FC<MenuProps> = ({
  open,
  activeIndex,
  onClose,
  onLogout,
  onLogin,
}) => {
  const {t} = useTranslation();

  return (
    <StyledDrawer
      closable={false}
      data-testid="drawer-navigation"
      key="drawer-menu"
      onClose={onClose}
      placement="left"
      title={<StyledNavBrand />}
      visible={open}
    >
      <StyledAntdMenu
        mobileVersion
        inlineCollapsed={false}
        mode="inline"
        selectedKeys={[activeIndex]}
      >
        <ItemLink
          key="1"
          icon={<SnippetsOutlined />}
          label={t('navbar.blog')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.blog.path}
        />
        {/* <ItemLink
          key="2"
          label={t('navbar.patients')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.patients.path}
        />
        <ItemLink
          key="3"
          label={t('navbar.orderAnalysis')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.orderAnalysis.path}
        /> */}

        <StyledAntdSubMenu
          icon={<UserOutlined />}
          key="mobile-sub-menu"
          title={<ProfileMenuTitle>{t('navbar.settings')}</ProfileMenuTitle>}
        >
          {/* <ItemLink
            key="4"
            label={t('navbar.profileAccount')}
            onClick={onClose}
            to={NAVIGATION_ROUTES.profileAccount.path}
          />
          <ItemLink
            key="5"
            label={t('navbar.profileSettings')}
            onClick={onClose}
            to={NAVIGATION_ROUTES.profileAccount.path}
          />
          <AntdMenu.Divider />
          <ItemLink
            key="6"
            label={t('navbar.profileHelp')}
            onClick={onClose}
            to={NAVIGATION_ROUTES.profileAccount.path}
          />
          <ItemLink
            key="7"
            label={t('navbar.profileSupport')}
            onClick={onClose}
            to={NAVIGATION_ROUTES.profileAccount.path}
          /> */}
          <StyledAntdMenu.Divider />
          <SubMenuItem label={t('navbar.logout')} onClick={onLogout} />
          <SubMenuItem label={t('navbar.login')} onClick={onLogin} />
        </StyledAntdSubMenu>
      </StyledAntdMenu>
    </StyledDrawer>
  );
};

export default Menu;
