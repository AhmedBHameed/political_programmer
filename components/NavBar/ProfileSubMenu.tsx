import {CaretDownOutlined, SnippetsOutlined} from '@ant-design/icons';
import {Menu} from 'antd';
import React from 'react';

import {useTranslation} from '../shared/hooks/useTranslation';
import SubMenuItem from './SubMenuItem';
import {ProfileMenuTitle} from './shared/ProfileMenuTitle';
import {StyledAntdMenu, StyledAntdSubMenu} from './shared/sharedStyles';

/**
 * TODO: fix Submenu bug @see https://github.com/ant-design/ant-design/issues/27176
 */
interface ProfileSubMenuProps {
  activeIndex: string;
  onClose?: () => void;
  onLogout?: () => void;
  onLogin?: () => void;
  className?: string;
}

const ProfileSubMenu: React.FC<ProfileSubMenuProps> = ({
  activeIndex,
  className,
  onClose,
  onLogin,
  onLogout,
}) => {
  const {t} = useTranslation();

  return (
    <StyledAntdMenu
      className={className}
      data-testid="profile-tablet-submenu"
      mode="horizontal"
      selectedKeys={[activeIndex]}
    >
      <StyledAntdSubMenu
        icon={<SnippetsOutlined />}
        key="profileMenu"
        title={
          <ProfileMenuTitle>
            {t('navbar.settings')} <CaretDownOutlined />
          </ProfileMenuTitle>
        }
      >
        {/* <ItemLink
          key="4"
          label={t('navbar.profileAccount')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.profileAccount.path}
        />
        <ItemLink
          key="5"
          label={t('navbar.profileSettings')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.profileAccount.path}
        />
        <Menu.Divider />
        <ItemLink
          key="6"
          label={t('navbar.profileHelp')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.profileAccount.path}
        />
        <ItemLink
          key="7"
          label={t('navbar.profileSupport')}
          onClick={onClose}
          to={NAVIGATION_ROUTES.profileAccount.path}
        /> */}
        <Menu.Divider />
        <SubMenuItem label={t('navbar.logout')} onClick={onLogout} />
        <SubMenuItem label={t('navbar.login')} onClick={onLogin} />
      </StyledAntdSubMenu>
    </StyledAntdMenu>
  );
};

export default ProfileSubMenu;
