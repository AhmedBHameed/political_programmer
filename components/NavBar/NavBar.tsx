import {SnippetsOutlined} from '@ant-design/icons';
import {Layout} from 'antd';
import React, {useCallback, useState} from 'react';

import styled from 'styled-components';
import NAVIGATION_ROUTES from '../../config/NavigationRoutes';

import devices from '../../styles/mediaQueries';
import {useTranslation} from '../shared/hooks/useTranslation';
import useViewport from '../WindowViewport/hooks/windowViewportHook';
import BurgerButton from './BurgerButton';
import useLogout from './hooks/logoutHook';
import useObtainActiveNavLink from './hooks/obtainActiveNavLinkHook';
import ItemLink from './ItemLink';
import MobileMenu from './MobileMenu';
import NavBrand from './NavBrand';
import ProfileSubMenu from './ProfileSubMenu';
import {spaceXSmall, StyledAntdMenu} from './shared/sharedStyles';

const {Header} = Layout;

/**
 * workaround solution see L:116
 */
const StyledProfileSubMenu = styled(ProfileSubMenu)`
  display: none;
  @media ${devices.mdMediaQuery} {
    display: block;
  }
`;

const StyledHeader = styled(Header)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: white;
  box-shadow: rgba(60, 64, 67, 0.3) 0px 1px 2px 0px,
    rgba(60, 64, 67, 0.15) 0px 1px 3px 1px;
  ${spaceXSmall}
`;

const ItemsContainer = styled.div`
  display: flex;
  height: 100%;
`;

const NavBar: React.FC = () => {
  const [openMenu, setOpenMenu] = useState(false);

  const {t} = useTranslation();
  const {handleLogout} = useLogout();
  const {activeIndex} = useObtainActiveNavLink();
  const {isTabletOrLarger} = useViewport();

  const openMobileMenu = useCallback(() => {
    setOpenMenu(true);
  }, [setOpenMenu]);

  const closeMobileMenu = useCallback(() => {
    setOpenMenu(false);
  }, [setOpenMenu]);

  return (
    <>
      <StyledHeader className="header">
        <NavBrand />
        <ItemsContainer>
          {isTabletOrLarger && (
            <StyledAntdMenu
              data-testid="tablet-navbar"
              mode="horizontal"
              selectedKeys={[activeIndex]}
            >
              <ItemLink
                key="1"
                icon={<SnippetsOutlined />}
                label={t('navbar.blog')}
                to={NAVIGATION_ROUTES.blog.path}
              />
              {/* <ItemLink
                key="2"
                label={t('navbar.patients')}
                to={NAVIGATION_ROUTES.patients.path}
              />
              <ItemLink
                key="3"
                label={t('navbar.orderAnalysis')}
                to={NAVIGATION_ROUTES.orderAnalysis.path}
              /> */}
            </StyledAntdMenu>
          )}

          <StyledProfileSubMenu
            activeIndex={activeIndex}
            onLogout={handleLogout}
          />
          {!isTabletOrLarger && <BurgerButton onClick={openMobileMenu} />}
        </ItemsContainer>
        {/*
          /**
              {isTabletOrLarger ? (
                <ProfileSubMenu
                  activeIndex={activeIndex}
                  onLogout={handleLogout}
                />
              ) : (
                <BurgerButton onClick={openMobileMenu} />
              )}
           * 
           */}

        {/*
         * TODO: This is a workaround till ant design team solve their issue
         * @see https://github.com/ant-design/ant-design/issues/27176#issuecomment-812059154
         * If they solve the issue, please use the code above.
         */}
      </StyledHeader>

      {!isTabletOrLarger && (
        <MobileMenu
          activeIndex={activeIndex}
          onClose={closeMobileMenu}
          onLogout={handleLogout}
          open={openMenu}
        />
      )}
    </>
  );
};

export default NavBar;
