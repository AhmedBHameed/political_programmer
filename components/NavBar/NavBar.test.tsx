import {act, fireEvent, render} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import {ThemeProvider} from 'styled-components';
import {mockLocationPathname, mockMediaQuery} from '../../setupTests';
import {md, sm} from '../../styles/mediaQueries';

import theme from '../../styles/Theme';
import ViewportProvider from '../WindowViewport/ViewportProvider';
import NavBar from './NavBar';

function renderComponent() {
  return render(
    <Router>
      <ViewportProvider>
        <ThemeProvider theme={theme}>
          <NavBar />
        </ThemeProvider>
      </ViewportProvider>
    </Router>
  );
}

describe('NavBar', () => {
  beforeEach(() => {
    mockLocationPathname.mockImplementation(() => ({
      pathname: '/',
    }));

    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    mockMediaQuery(md, md);

    const {container} = renderComponent();

    expect(container).toMatchSnapshot();
  });

  describe('Viewport behaviour', () => {
    it('shows tablet navbar version on screen bigger or equal to tablet size', async () => {
      mockMediaQuery(md, md);

      const {queryByTestId} = renderComponent();

      expect(await queryByTestId('tablet-navbar')).toBeInTheDocument();
      expect(await queryByTestId('profile-tablet-submenu')).toBeInTheDocument();
      expect(await queryByTestId('burger-button')).not.toBeInTheDocument();
    });

    it('shows mobile navbar version on screen smaller than tablet size', async () => {
      mockMediaQuery(md, sm);
      const {queryByTestId} = renderComponent();

      expect(await queryByTestId('tablet-navbar')).not.toBeInTheDocument();
      // expect(
      //   await queryByTestId('profile-tablet-submenu')
      // ).not.toBeInTheDocument();
      // TODO: Please check the comment in `NavBar.tsx component L:116`
      expect(await queryByTestId('profile-tablet-submenu')).toBeInTheDocument();
      expect(await queryByTestId('burger-button')).toBeInTheDocument();
    });

    it('shows drawer menu on mobile version when burger button clicked', async () => {
      mockMediaQuery(md, sm);

      const {getByTestId} = renderComponent();

      act(() => {
        fireEvent.click(getByTestId('burger-button'));
      });

      expect(getByTestId('drawer-navigation')).toBeInTheDocument();
    });
  });
});
