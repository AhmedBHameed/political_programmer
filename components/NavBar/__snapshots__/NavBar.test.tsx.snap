// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<div>
  <header
    class="ant-layout-header sc-iqsfdx hqwVIt header"
  >
    <div
      class="sc-iCfLBT ljtVXz"
    >
      <a
        href="/"
      >
        <svg
          class="sc-gKckTs jaWLCM"
        >
          brand.svg
        </svg>
      </a>
    </div>
    <div
      class="sc-crHlIS dgufeY"
    >
      <ul
        class="ant-menu ant-menu-light sc-dkPtyc efIkys ant-menu-root ant-menu-horizontal"
        data-testid="tablet-navbar"
        role="menu"
      >
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item sc-eCImvq bkELLT ant-menu-item-selected"
          data-testid="navbar-item"
          role="menuitem"
        >
          <span
            aria-label="dashboard"
            class="anticon anticon-dashboard ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="dashboard"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span>
            <a
              class="sc-hKwCoD hsSRrP"
              href="/"
            >
              navbar.dashboard
            </a>
          </span>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item sc-eCImvq bkELLT"
          data-testid="navbar-item"
          role="menuitem"
        >
          <svg
            class="ant-menu-item-icon"
          >
            patient.svg
          </svg>
          <span>
            <a
              class="sc-hKwCoD hsSRrP"
              href="/patients"
            >
              navbar.patients
            </a>
          </span>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item sc-eCImvq bkELLT"
          data-testid="navbar-item"
          role="menuitem"
        >
          <span
            aria-label="shopping-cart"
            class="anticon anticon-shopping-cart ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="shopping-cart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 00-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 100 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 00-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 00-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6z"
              />
            </svg>
          </span>
          <span>
            <a
              class="sc-hKwCoD hsSRrP"
              href="/order-analysis"
            >
              navbar.orderAnalysis
            </a>
          </span>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="visibility: hidden; position: absolute; display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
    </div>
    <ul
      class="ant-menu ant-menu-light sc-dkPtyc efIkys sc-kDThTU jrinWD ant-menu-root ant-menu-horizontal"
      data-testid="profile-tablet-submenu"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal sc-jrQzUz lpjTkG"
        role="menuitem"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
          <span>
            <span
              class="sc-furvIG cGmXbe"
            >
              navbar.profile
               
              <span
                aria-label="caret-down"
                class="anticon anticon-caret-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="caret-down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </header>
</div>
`;
